### ISSUE TITLE
Error while installing php72w-pecl-imagick package

### ISSUE THREAD

**Issue by sharkydmm** [webtatic-rpms/php72w#17]
Hello, I have a similar problem as #16 , but it's not the same, that's why creating a new ticket instead or writing a comment in the other issue. Since today I have a problem with installing php72w-pecl-imagick package, without changing anything. When trying to install it, I'm receiving this error, while trying to build the Dockerfile.

```
...
---> Package php72w-pecl-imagick.x86_64 0:3.4.3-1.2.w7 will be installed
--> Processing Dependency: libMagickWand.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
--> Processing Dependency: libMagickCore.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
--> Finished Dependency Resolution
 Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (webtatic)
            Requires: libMagickWand.so.5()(64bit)
 Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (webtatic)
            Requires: libMagickCore.so.5()(64bit)
  You could try using --skip-broken to work around the problem
  You could try running: rpm -Va --nofiles --nodigest
 The command '/bin/sh -c rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm  && rpm -Uvh https://mirror.webtatic.com/yum/el7/webtatic-release.rpm  && yum install -y which bzip2 sudo nginx php72w-devel php72w-fpm php72w-bcmath php72w-cli php72w-common php72w-fpm php72w-gd php72w-gmp php72w-intl php72w-json php72w-mbstring php72w-mcrypt php72w-mysqlnd php72w-pdo php72w-pear php72w-pecl-igbinary php72w-pecl-imagick php72w-pecl-redis php72w-pecl-zip php72w-process php72w-recode php72w-soap php72w-tidy php72w-xml php72w-pecl-geoip php72w-pecl-zendopcache php72w-pecl-xdebug wget && yum group install -y "Development Tools" && yum clean all  && pecl install pcov  && pecl clear-cache  && touch /etc/php.d/pcov.ini  && echo "extension=pcov.so" > /etc/php.d/pcov.ini' returned a non-zero code: 1
 ERROR: Job failed: exit code 1
```
for reference, this is one of the builds from yesterday, where everything was fine
```
...
 ---> Package php72w-pecl-imagick.x86_64 0:3.4.3-1.2.w7 will be installed
 --> Processing Dependency: libMagickWand.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
 --> Processing Dependency: libMagickCore.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
...
```

Thank you in advance,
Have a nice day,
Martin

**Comment by HigH-HawK** [comment]
Hey Martin,

it looks like you already have the newest version of ImageMagick installed, hence the above error appears because the newest release does not seem to have the **libMagickWand.so.5** anymore.

Can you please run the following command and post the output:

```
yum --showduplicates list ImageMagick | expand
```

**Comment by sharkydmm** [comment]
Hey Chris,

Unfortunately, I can't give you an output from this command, since few days after I saw the problem is still present and I didn't have an answer how to fix it, I decided to switch to another  package provider, since my deployment was blocked, because of this issue.

But to be able to test it by yourself, here you are the command that was throwing the error

```
Step 5/16 : RUN rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm  && rpm -Uvh https://mirror.webtatic.com/yum/el7/webtatic-release.rpm  && yum install -y which bzip2 sudo nginx php72w-devel php72w-fpm php72w-bcmath php72w-cli php72w-common php72w-fpm php72w-gd php72w-gmp php72w-intl php72w-json php72w-mbstring php72w-mcrypt php72w-mysqlnd php72w-pdo php72w-pear php72w-pecl-igbinary php72w-pecl-imagick php72w-pecl-redis php72w-pecl-zip php72w-process php72w-recode php72w-soap php72w-tidy php72w-xml php72w-pecl-geoip php72w-pecl-zendopcache php72w-pecl-xdebug wget && yum group install -y "Development Tools" && yum clean all  && pecl install pcov  && pecl clear-cache  && touch /etc/php.d/pcov.ini  && echo "extension=pcov.so" > /etc/php.d/pcov.ini
``` 

and the failing response in the end of the build

```
... (more lines for installing dependencies)
--> Processing Dependency: groff-base for package: perl-Pod-Perldoc-3.20-4.el7.noarch
 ---> Package perl-podlators.noarch 0:2.5.1-3.el7 will be installed
 ---> Package php72w-pecl-imagick.x86_64 0:3.4.3-1.2.w7 will be installed
 --> Processing Dependency: libMagickWand.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
 --> Processing Dependency: libMagickCore.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
 --> Running transaction check
 ---> Package groff-base.x86_64 0:1.22.2-8.el7 will be installed
 ---> Package perl-HTTP-Tiny.noarch 0:0.033-3.el7 will be installed
 ---> Package perl-parent.noarch 1:0.225-244.el7 will be installed
 ---> Package php72w-pecl-imagick.x86_64 0:3.4.3-1.2.w7 will be installed
 --> Processing Dependency: libMagickWand.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
 --> Processing Dependency: libMagickCore.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
 --> Finished Dependency Resolution
 Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (webtatic)
            Requires: libMagickWand.so.5()(64bit)
 Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (webtatic)
            Requires: libMagickCore.so.5()(64bit)
  You could try using --skip-broken to work around the problem
  You could try running: rpm -Va --nofiles --nodigest
 The command '/bin/sh -c rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm  && rpm -Uvh https://mirror.webtatic.com/yum/el7/webtatic-release.rpm  && yum install -y which bzip2 sudo nginx php72w-devel php72w-fpm php72w-bcmath php72w-cli php72w-common php72w-fpm php72w-gd php72w-gmp php72w-intl php72w-json php72w-mbstring php72w-mcrypt php72w-mysqlnd php72w-pdo php72w-pear php72w-pecl-igbinary php72w-pecl-imagick php72w-pecl-redis php72w-pecl-zip php72w-process php72w-recode php72w-soap php72w-tidy php72w-xml php72w-pecl-geoip php72w-pecl-zendopcache php72w-pecl-xdebug wget && yum group install -y "Development Tools" && yum clean all  && pecl install pcov  && pecl clear-cache  && touch /etc/php.d/pcov.ini  && echo "extension=pcov.so" > /etc/php.d/pcov.ini' returned a non-zero code: 1
 ERROR: Job failed: exit code 1
```

Sorry for not being able, to help you investigate!

**Comment by HigH-HawK** [comment]
Hey Martin,

I have the same issue but since this is a productive environment, I need to put a bit more work into it to change the package without disrupting the business. Hopefully the creators can fix this until then, I will have a look and setup a new machine and try moving everything over.

**Comment by YaroKasear** [comment]
I am also unable to install this package.

Results from yum --showduplicates list ImageMagick | expand

```
Loaded plugins: fastestmirror, langpacks, product-id, search-disabled-repos, subscription-
              : manager

This system is not registered with an entitlement server. You can use subscription-manager to register.

Loading mirror speeds from cached hostfile
 * base: mirror.dal.nexril.net
 * epel: mirror.compevo.com
 * extras: centos.s.uw.edu
 * nux-dextop: mirror.li.nux.ro
 * updates: mirror.vcu.edu
 * webtatic: uk.repo.webtatic.com
Resolving Dependencies
--> Running transaction check
---> Package php72w-pecl-imagick.x86_64 0:3.4.3-1.2.w7 will be installed
--> Processing Dependency: libMagickWand.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
--> Processing Dependency: libMagickCore.so.5()(64bit) for package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64
--> Finished Dependency Resolution
Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (webtatic)
           Requires: libMagickWand.so.5()(64bit)
Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (webtatic)
           Requires: libMagickCore.so.5()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

Results from yum --showduplicates list ImageMagick | expand

```
Loaded plugins: fastestmirror, langpacks, product-id, search-disabled-repos,
              : subscription-manager
Loading mirror speeds from cached hostfile
 * base: mirror.dal.nexril.net
 * epel: mirror.dal.nexril.net
 * extras: centos.den.host-engine.com
 * nux-dextop: mirror.li.nux.ro
 * updates: mirrors.gigenet.com
 * webtatic: us-east.repo.webtatic.com
Installed Packages
ImageMagick.x86_64                     6.9.10.68-3.el7                     @base
Available Packages
ImageMagick.i686                       6.9.10.68-3.el7                     base
ImageMagick.x86_64                     6.9.10.68-3.el7                     base
```

It's not a huge deal for me except Nextcloud really wants me to install this module and because of this issue I can't. If I were using Arch Linux I could just rebuild the package, but I can't find a way to build an arbitrary RPM from source without doing it all from scratch in CentOS. Perhaps this module needs a rebuild?

**Comment by legecha** [comment]
Hi, I have this same problem so I didn't want to open up a new issue.

```
~ yum --showduplicates list ImageMagick | expand
Loaded plugins: fastestmirror, replace
Loading mirror speeds from cached hostfile
 * base: mirrors.ukfast.co.uk
 * epel: mirrors.ukfast.co.uk
 * extras: mirrors.ukfast.co.uk
 * updates: mirrors.ukfast.co.uk
 * webtatic: uk.repo.webtatic.com
Installed Packages
ImageMagick.x86_64                     6.7.8.9-18.el7                      @base
Available Packages
ImageMagick.i686                       6.9.10.68-3.el7                     base
ImageMagick.x86_64                     6.9.10.68-3.el7                     base
```

It appears I don't have a duplicate and the upgrade is legit, but presumably the newest version of ImageMagick doesn't provide the necessary core and wand libs?

Here's a snippet of `yum update`:

```
--> Finished Dependency Resolution
Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (@webtatic)
           Requires: libMagickCore.so.5()(64bit)
           Removing: ImageMagick-6.7.8.9-18.el7.x86_64 (@base)
               libMagickCore.so.5()(64bit)
           Updated By: ImageMagick-6.9.10.68-3.el7.x86_64 (base)
               Not found
Error: Package: php72w-pecl-imagick-3.4.3-1.2.w7.x86_64 (@webtatic)
           Requires: libMagickWand.so.5()(64bit)
           Removing: ImageMagick-6.7.8.9-18.el7.x86_64 (@base)
               libMagickWand.so.5()(64bit)
           Updated By: ImageMagick-6.9.10.68-3.el7.x86_64 (base)
               Not found
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
```

Any suggestions?

**Comment by HigH-HawK** [comment]
I may have found a solution for this problem and it seems to work.

Requirements:

- php72w-devel
- gcc
- ImageMagick
- ImageMagick-devel

First we remove everything imagick and ImageMagick related:

```
#: pecl uninstall imagick
#: yum remove ImageMagick ImageMagick-devel -y
```

Now we install all the requirements from above:

```
#: yum check-update; yum install -y php72w-devel gcc ImageMagick ImageMagick-devel
```

Install the imagick lib again: (if there are prompts to enter something, leave them at "autodetect" and hit Enter)

```
#: pecl install imagick
```

PECL will now compile and install imagick again - Please don't worry if it comes up with an "error" that it couldn't add the extension to the php.ini - You can do this yourself with the following command:

```
#: echo "extension=imagick.so" > /etc/php.d/imagick.ini
```

Last but not least, the Apache needs to be restarted:

```
#: systemctl restart httpd
```

**Comment by andytson** [comment]
please try the latest updates, which have been in the testing repository, but now in the main repository

**Comment by coolgromm** [comment]
please try 

```
yum install -y ImageMagick-devel
yum -y install php72w-pecl-imagick
```

**Comment by HigH-HawK** [comment]
> please try the latest updates, which have been in the testing repository, but now in the main repository

Unfortunately I can't try this anymore, since I already have done the steps I mentioned above, so it's working for me again.

> please try
> 
> ```
> yum install -y ImageMagick-devel
> yum -y install php72w-pecl-imagick
> ```

The package `ImageMagick-devel` was already installed - `php72w-pecl-imagick` was already installed but had issues during the CentOS update procedure, due to a change in ImageMagick.

Again, I could solve it like I described above and it is working perfectly fine for me.

**Comment by andytson** [comment]
`yum install php72w-pecl-imagick` or `yum update php72w-pecl-imagick` should be enough if you are running RHEL/CentOS 7.8. Any other version (such as a pinned version 7 < 7.8) 

I'm closing this issue as it is resolved now and locking it as there should be no need to manually build the extension now
